File: chat-backend/src/main/java/model/Group.java
package model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
import java.util.UUID;

@Document(collection = "group")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Group {
    private String groupId;

    private String groupName;

    private String adminId;

    private List<Integer> userIds;

    public Group(String groupName,String adminId) {
        this.groupId = UUID.randomUUID().toString();
        this.groupName = groupName;
        this.adminId = adminId;
    }


}

File: chat-backend/src/main/kotlin/com/thm/greenhat/greenhatchat/model/Group.kt
package com.thm.greenhat.greenhatchat.model

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.mongodb.core.mapping.Document


@Document("group")
data class Group(
        @Id var _id: String = ObjectId.get().toString(),
        var name:String,
        var admin:String,
        var users:MutableList<User> = mutableListOf()
)
{
    @Transient
    var messages: MutableList<Message> = mutableListOf()
}